Test/Resolve-WhiskeyTaskPathParameter.Tests.ps1

```diff
@@ -358,8 +408,188 @@ Describe ('Resolve-WhiskeyTaskPathParameter.when a path uses different case to t
+    }
+}
+
+Describe 'Resolve-WhiskeyTaskPath.when path does not exist and some paths use wildcards and others do not' {
+    It 'should normalize directory separators' {
+        Init
+        WhenRunningTask 'ValidateMandatoryNonexistentFilesTask' -Parameter @{ 'Path' = @( 'file.txt','*.json','anotherfile.txt') }
+        $expectedPaths = & {
+            Resolve-RelativePath 'file.txt'
+            Resolve-RelativePath 'anotherfile.txt'
+        }
+        ThenTaskCalled -WithParameter @{ 'Path' = $expectedPaths }
+    }
+}
+
+Describe 'Resolve-WhiskeyTaskPath.when path does not exist and user is using directory separator from another OS' {
+    It 'should normalize directory separators' {
+        if( $IsWindows )
+        {
+            $separator = '/'
+        }
+        else
+        {
+            $separator = '\'
+        }
+        Init
+        $paths = @(
+            ((Join-Path -Path $testRoot -ChildPath 'some\custom\path.zip') -replace ([regex]::Escape([IO.Path]::DirectorySeparatorChar)),$separator),
+            ('some{0}other{0}path.zip' -f $separator)
+            # Make sure on Windows, separator gets switched to backslash.
+            'another{0}path.txt' -f [IO.Path]::AltDirectorySeparatorChar
+        )
+        WhenRunningTask 'ValidateMandatoryNonexistentFilesTask' -Parameter @{ 'Path' = $paths }
+        $expectedPaths = & {
+            Resolve-RelativePath ('some{0}custom{0}path.zip' -f [IO.Path]::DirectorySeparatorChar)
+            Resolve-RelativePath ('some{0}other{0}path.zip' -f [IO.Path]::DirectorySeparatorChar)
+            Resolve-RelativePath ('another{0}path.txt' -f [IO.Path]::DirectorySeparatorChar)
+        }
+        ThenTaskCalled -WithParameter @{ 'Path' = $expectedPaths }
+    }
+}
+
+Describe 'Resolve-WhiskeyTaskPath.when using glob syntax' {
+    AfterEach { Reset }
+    It 'should return files matching globs' {
+        Init
+        GivenFile 'absolute','abc.yml', 'cdf.yml', 'root.txt','root.txt.orig', 'dir\file.txt', 'dir\whiskey.yml', 'dir1\file.txt.orig', 'dir2\dir3\anotherfile.txt', 'dir2\dir3\anotherfile.txt.fubar'
+        # These files test that we search hidden places
+        GivenFile '.hidden.yml' -Hidden
+        GivenDirectory '.hidden' -Hidden
+        GivenFile '.hidden\config.txt'
+        # These files test that we're doing a case-senstive search on Linux.
+        GivenFile 'uppercase.TXT'
+
+        WhenRunningTask 'ValidatePathWithGlobTask' -Parameter @{ 
+            'Path' = @( '*.yml', '**\*.txt', '**\*.txt.*', (Join-Path -Path $testRoot -ChildPath 'absolute')); 
+            'Exclude' = @( 'PSModules\**', '**\*.orig', 'whiskey.yml' ) }
+        $expectedPaths = & {
+            Resolve-RelativePath 'absolute'
+            Resolve-RelativePath 'abc.yml'
+            Resolve-RelativePath 'cdf.yml'
+            Resolve-RelativePath 'root.txt'
+            Resolve-RelativePath 'dir\file.txt'

```

Comment:
After work put in windows-vs-non above, these seem "\"-specific.

Reply:
The `Resolve-RelativePath` function converts them to the correct separator for the current platform.

Reply:
The previous test is there to explicitly make sure separators from other OSes sent in by users are handled properly. That way we don't have false positives because all our tests used the same separator.